react/components/RenderProvider.tsx

```diff
@@ -762,9 +764,16 @@ class RenderProvider extends Component<Props, RenderProviderState> {
 
   public fetchComponents = async (
     components: RenderRuntime['components'],
-    extensions: RenderRuntime['extensions']
+    extensions?: RenderRuntime['extensions']
   ) => {
     const { runtime } = this.props
+    if (!extensions) {
+      const componentsNames = Object.keys(components)
+      extensions = componentsNames.reduce(
+        (acc, component) => ({ ...acc, [component]: { component } }),
```

Comment:
⛏ I'd suggest doing this instead; it's a bit less tidy but it should have better performance and lower memory impact
```suggestion
        (acc, component) => {
          acc[component] = { component }
          return acc
        }
```